lib/msf/base/sessions/command_shell.rb

```diff
@@ -92,6 +92,27 @@ def bootstrap(datastore = {}, handler = nil)
     session = self
 
     if datastore['AutoVerifySession']
+      session_info = ''
+
+      # Read the initial output and mash it into a single line
+      # Timeout set to 1 to read in banner of all payload responses (may capture prompt as well)
+      # Encoding is not forced to support non ASCII shells
+      if session.info.nil? or session.info.empty?
+        banner = shell_read(-1, 1)
+        if banner && !banner.empty?
+          banner.gsub!(/[\x00-\x08\x0b\x0c\x0e-\x19\x7f-\xff]+/n, "_")
+          banner.strip!
+
+          banner = """
+Shell Banner:
+#{banner}
+-----
+          """
```

Comment:
> I don't think triple quote is something ruby supports

Can confirm, Ruby doesn't support triple quotes in the same way that other languages supports triple quotes :+1:

When you do something like this in Ruby:
```ruby
x = """
  hello world
"""
```

It's closer to C's handling of multiline strings, so this is basically the same as:
```ruby
x = "" "\n    hello world\n" ""
```

Although i'm not sure we'd want a banner variable wrapped like this either way, it might be unexpected behavior for users

Reply:
Sorry guys, that was my Python brain pulling a sneaky on me 😅